src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/tests/EnvironmentVariablesTest.cs

```diff
@@ -331,14 +308,17 @@ void ReloadLoop()
 
                 _ = Task.Run(ReloadLoop);
 
-                while (!cts.IsCancellationRequested)
```

Comment:
The problem was that the token can be cancelled by the time we get here and options will stay null. We need to make sure to call `options = config.Get<MyOptions>()` at least once.